src/shared/components/oncoprint/OncoprintUtils.ts

```diff
@@ -752,3 +949,37 @@ export function makeGenesetHeatmapTracksMobxPromise(
         default: []
     });
 }
+
+export function extractTreatmentSelections(text:string, selectedTreatments:ISelectOption[], treatmentsMap:{[treatmentId:string]:ISelectOption}):string {
+
+    // get values from input string
+    const elements = splitHeatmapTextField(text);
+
+    // check values for valid treatment ids
+    const selectedKeys:string[] = _.map(selectedTreatments,'value');
+    const detectedTreatments:string[] = [];
+    _.each(elements, (d:string)=> {
+        if (d in treatmentsMap) {
+            detectedTreatments.push(d);
+            if (! selectedKeys.includes(d)) {
+                selectedTreatments.push( treatmentsMap[d] );
+            }
+        }
+    });
+
+    // remove valid treatment ids from the input string
+    if (detectedTreatments.length > 0) {
+        _.each(detectedTreatments, (d:string) => {
+            text = text.replace(d, "");
+        });
+    }
+
+    // return the input string
+    return text.trim().replace("\t+","\t").replace(" +"," ");
```

Comment:
can you explain what the purpose is of these changes, and of returning this string?

Reply:
Ok. The text area component has editable text in its text area element. This text is updated when a word identical to a treatment identifier is recognized and the word is removed from the text. Normally I would do this with a observable MobX field in OncoprintControls. However, I was unable to get this to work for reasons unknown to me. Instead the onChange handler function in TextIconArea component can handle a return value:

```
 onChangeTextArea?: (textAreaContents:string) => string;
```

The TextIconArea uses the string returned by:
```
 return text.trim().replace("\t+","\t").replace(" +"," ");
```

to update the text area.

Reply:
Ah okay thanks, I think thats a good solution